src/kubeobjects/certificates.go

```diff
@@ -0,0 +1,23 @@
+package kubeobjects
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"time"
+
+	"github.com/go-logr/logr"
+)
+
+func ValidateCertificateExpiration(certData []byte, renewalThreshold time.Duration, now time.Time, log logr.Logger) (bool, error) {
+	if block, _ := pem.Decode(certData); block == nil {
+		log.Info("failed to parse certificate", "error", "can't decode PEM file")
+		return false, nil
+	} else if cert, err := x509.ParseCertificate(block.Bytes); err != nil {
+		log.Info("failed to parse certificate", "error", err)
+		return false, err
+	} else if now.After(cert.NotAfter.Add(-renewalThreshold)) {
```

Comment:
why don't you use `time.Now()` like you did in the first commit ?
so you don't have to pass in `now` all those times

Reply:
that means the time is the same for the whole execution
using `time.Now()` again could lead to unintended effects